### PR TITLE
add pick task json output node

### PR DESCRIPTION
**Pull request by knorth55**
pick task for #1632

### CODE REVIEW

**Comment by knorth55** [comment]
ping

File: jsk_2016_01_baxter_apc/node_scripts/output_json_pick.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        self.arm_target_bin['left'] = \
+            rospy.get_param('/left_hand/target_bin').replace(':', '')
+
+    def update_data(self):
+        is_updated = False
+        for arm in ['right', 'left']:
+            target_bin = self.arm_target_bin[arm]
+            if target_bin not in self.finished_bin_list \
+                    and target_bin in 'abcdefghijkl' \
+                    and self.arm_state[arm] == ":place_object":
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/node_scripts/output_json_pick.py]: Is this true?
I thought it is `"place_object"`..

File: jsk_2016_01_baxter_apc/launch/main.launch [hunk 2]
```diff
@@ -16,6 +16,12 @@
     </rosparam>
   </node>
 
+  <node pkg="jsk_2016_01_baxter_apc" type="output_json_pick.py" name="output_json_pick" respawn="true">
+    <rosparam subst_value="true">
+        json: $(arg json)
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/launch/main.launch]: should be two spaces.

**Comment by wkentaro** [comment]
Please how the output looks like.

File: jsk_2016_01_baxter_apc/node_scripts/output_json_pick.py (hunk 1 above)
Comment (knorth55) [inline jsk_2016_01_baxter_apc/node_scripts/output_json_pick.py]: that's true , thank you

**Comment by knorth55** [comment]
```
$ roscore
$ rosparam set /left_hand/target_bin :a
$ rosparam set /left_hand/state place_object
$ rosparam set /right_hand/state place_object
$ rosparam set /right_hand/target_bin :b
```

**Comment by wkentaro** [comment]
Are there any validation tools for the output which is official?

**Comment by knorth55** [comment]
i could not find any tool.
In the official rule, we only need to tell correct position of objects when the trial finished.

**Comment by wkentaro** [comment]
Then could you please ask about it in the mailing list?

2016年6月11日土曜日、Shingo Kitagawa<notifications@github.com>さんは書きました:

> i could not find any tool.
> In the official rule, we only need to tell correct position of objects
> when the trial finished.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/start-jsk/jsk_apc/pull/1661#issuecomment-225226642>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEHFk9A7nwqvrh9Iq7kPEyrrprJZVO8oks5qKY0lgaJpZM4IyFJq>
> .
>


-- 
和田 健太郎 / Kentaro Wada
http://wkentaro.com

**Comment by knorth55** [comment]
what do you mean by validation?
they say that if output json is wrong, we get penalty.
Do you want to know the penalty point?

**Comment by wkentaro** [comment]
> what do you mean by validation?

I mean the check for the format of the output json file.
My question is whether below json file format is official or not:

```
{
    "bin_contents": {
        "bin_A": [
            "creativity_chenille_stems"
        ], 
        ...
        "bin_L": [
            "ticonderoga_12_pencils", 
            "expo_dry_erase_board_eraser", 
            "expo_dry_erase_board_eraser", 
            "dr_browns_bottle_brush"
        ]
    }, 
    "tote_contents": [
        "up_glucose_bottle", 
        ...
        "kleenex_tissue_box"
    ], 
    "work_order": [
        {
            "bin": "bin_A", 
            "item": "kleenex_tissue_box"
        }, 
        ...
        {
            "bin": "bin_L", 
            "item": "dr_browns_bottle_brush"
        }
    ]
}
```

**Comment by wkentaro** [comment]
> Do you want to know the penalty point?

Please ask about it also.

**Comment by knorth55** [comment]
so you want to know output json format.
there is no information about its format, and i will send an e-mail

**Comment by wkentaro** [comment]
Please ask about the demerit points as the penalty also.

**Comment by knorth55** [comment]
what is demerit point?

**Comment by knorth55** [comment]
ok i got it

**Comment by knorth55** [comment]
fixed to use `rospy.Timer`

**Comment by wkentaro** [comment]
👍 
BTW, could you please stop amending the commit next time as much as possible?
It makes the review hard..

**Comment by knorth55** [comment]
I will try my best

**Pull request merged**